### PR TITLE
Fix Torch visualization of color image layers

### LINKED ISSUE #721: Image visualization
With torch, images are fed into the network as BGR (I assume), but RGB convention when displaying intermediate results, so intermediate results show up in false colors.

To show the issue, I created a network with just one multiply(1) layer, here's the inference result:

lenabgrrgb

### PR DESCRIPTION
**Pull request by gheinrich**
close #721

### CODE REVIEW

File: digits/utils/image.py [hunk 1]
```diff
@@ -306,6 +307,7 @@ def get_layer_vis_square(data,
     normalize -- whether to normalize the data when visualizing
     max_width -- maximum width for the vis_square
     """
+    assert channel_order == 'RGB' or channel_order == 'BGR'
```
Comment (lukeyeager) [inline digits/utils/image.py]: How about:
```python
if channel_order not in ['RGB', 'BGR']:
  raise ValueError('Unsupported channel_order %s' % channel_order)
```
> Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.
https://docs.python.org/2/library/exceptions.html#exceptions.ValueError

I've got `asserts` throughout the code, but I'm trying to phase them out where appropriate.

Comment (gheinrich) [inline digits/utils/image.py]: OK no problem I'll make a mental note not to use asserts :-)

**Comment by lukeyeager** [comment]
Since RGB is the more standard representation (OpenCV users excepted), can we make that the default for `get_layer_vis_square`? Caffe should need to provide the extra flag, not Torch.

This is obviously not a big deal, but will make integrating other frameworks slightly more straightforward.

**Comment by pansk** [comment]
Wouldn't switching to a different representation require changing the way how the image is fed to the network, and thus break previously trained networks?
In this case a conversion tool (which swaps weights and biases for R she B layers) could be quite useful, but I really don't know how to determine automatically if a layer is receiving/producing a RGB image.

**Comment by gheinrich** [comment]
> Wouldn't switching to a different representation require changing the way how the image is fed to the network

This patch doesn't change the way data are fed to the network. The network (for Torch) still receives data in RGB order.

**Comment by gheinrich** [comment]
> Since RGB is the more standard representation (OpenCV users excepted), can we make that the default for get_layer_vis_square?

I thought so too but didn't dare to change the Caffe path :-)

I have updated the commit to make RGB the default channel order. Thanks for the review!

@pansk could you try 395dbb3 ? Thanks!

**Comment by pansk** [comment]
Sorry for the misunderstanding, I thought Luke was also referring to how the network is fed.
The solution is perfect, thank you!

**Comment by gheinrich** [comment]
Thanks for the feedback. I'll merge the change then.

**Pull request merged**